### PR TITLE
[VBLOCKS-2829] Update to SDK version rc23

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] ~The `README.md` reflects new **features**~
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] ~New unit tests and integration tests have beed added~
 - [x] Code coverage is more or equal to the previous coverage
 - [x] A visual inspection of the `Files changed` tab was made prior to submitting the pull request ensuring the style guide was followed
 - [x] [CI pipeline](https://app.circleci.com/pipelines/github/twilio/twilio-voice-react-native-app) is green
 - [x] ~Included screenshot if necessary~

## Description

This PR updates the SDK version to rc22 and uses the new CallInvite event API. These are all under the hood changes and should not affect the UX of the end-user.

### Breakdown

- Update to SDK version RC22.
- Use new callinvite event api.
- Remove voice callinvite event api.

## Validation

- Manual testing the following flows:

```
[x] android
  [x] incoming call
    [x] native accept
      [x] local disconnect
      [x] remote disconnect
    [x] native reject
    [x] js accept
      [x] local disconnect
      [x] remote disconnect
    [x] js reject
    [x] cancel
      [x] home screen
      [x] incoming call screen
  [x] outgoing call
    [x] accept
      [x] local disconnect
      [x] remote disconnect
    [x] reject
    [x] cancel

[x] ios
  [x] incoming call
    [x] native accept
      [x] local disconnect
      [x] remote disconnect
    [x] native reject
    [x] cancel
  [x] outgoing call
    [x] accept
      [x] local disconnect
      [x] remote disconnect
    [x] reject
    [x] cancel
```

## Additional Notes

N/A

## Contributing to Twilio

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: app/src/store/bootstrap.ts [hunk 1]
```diff
@@ -114,8 +114,74 @@ export const bootstrapCallInvites = createTypedAsyncThunk<
...
+       * Handle the call invite accepted event.
+       */
+      const handleCallInviteAccepted = (call: TwilioCall) => {
+        // dispatch the new call
+        dispatch(handleCall({ call }));
+        handleSettledCallInvite(callInvite);
+
+        const callSid = callInvite.getCallSid();
+        const currentRoute = getCurrentRoute();
+        if (currentRoute?.name !== 'Incoming Call') {
```
Comment (charliesantos) [inline app/src/store/bootstrap.ts:128]: Is it easy to put these route names in an enum?

Comment (mhuynh5757) [reply app/src/store/bootstrap.ts:128]: I did the changes locally and they touch _a lot_ of files, are you sure you want to include them in this PR?

Comment (charliesantos) [reply app/src/store/bootstrap.ts:128]: Uggh. nvm then.

**Review by charliesantos — APPROVED** [review]
+1. Not sure why build is failing though..

**Review by kpchoy — APPROVED** [review]

File: app/src/store/bootstrap.ts (hunk 1 above)
Comment (mhuynh5757) [reply app/src/store/bootstrap.ts:128]: Ok, I'll shelf it for now and file a ticket.

**Comment by mhuynh5757** [comment]
I mentioned over Slack yesterday that a bug in the native Android layer is causing the CI failures, but @afalls-twilio just merged that fix to main. We're probably going to want to release rc23, update this PR, and then merge.

File: app/src/store/bootstrap.ts (hunk 1 above)
Comment (mhuynh5757) [reply app/src/store/bootstrap.ts:128]: Here's the ticket https://issues.corp.twilio.com/browse/VBLOCKS-2844

**Review by charliesantos — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,8 +5,19 @@
 ### Changes
 * Refactored and simplified navigators. Now, movement between screens is more
 explicit and has finer control.
-* Upgraded Twilio Voice SDK to `1.0.0-beta.4`.
+* Upgraded Twilio Voice SDK to `1.0.0-rc22`.
```
Comment (kpchoy) [inline CHANGELOG.md:8]:
```suggestion
* Upgraded Twilio Voice SDK to `1.0.0-rc23`.
```

**Review by kpchoy — CHANGES_REQUESTED** [review]

**Review by kpchoy — APPROVED** [review]

**Pull request merged**